### DISCUSSION TITLE
Wikidata features for the next release (2021)

### DISCUSSION THREAD

**Discussion by ross-spencer** [richardlehane/siegfried#159] [General]
A couple of thoughts about code to organize before the next release, roughly in order as I'd like to get to them, but we should discuss.

* [x] Translations [SPARQL issue](https://github.com/richardlehane/siegfried/issues/153) (Bug).
* [x] Provenance/reference [display in Wikidata results](https://github.com/ross-spencer/WikiDP-Issues/issues/4) (Sustainability).
* [x] Combining [Wikibase provenance with Wikidata](https://github.com/richardlehane/siegfried/pull/157) (Feature PR).
* [ ] ~~Output software links [with Wikidata results](https://github.com/richardlehane/siegfried/pull/152) (Feature PR)~~.

What do others think?

**Comment by richardlehane** [comment]
Looks good to me. I've got to do some infrastructure stuff over the next month, especially because of Bintray EOL.

- [ ] Replace bintray for debian builds
- [ ] Explore Github Actions to replace travis, appveyor and bintray

**Reply by ross-spencer** [reply]
Nice. Have you the end of the month in mind for some sort of infra-release? Or a little bit further away in mind? I'm not sure how much I can get done, but I'd work through each of these in order, so would love to figure out that translation bug asap (haven't quite solved it yet) and depending on what you'd like to do with versioning, removing one of the provenance field options to simplify things would be great. With a little bit longer, that Wikibase provenance idea feels within reach. The Software links idea seems a bit further away and worth some more discussion.

**Reply by richardlehane** [reply]
the bintray EOL date is 1 May, so I need to get it swapped out by then. Maybe mid-April as a target date??

**Reply by ross-spencer** [reply]
That works! I was a bit worried about the translation one but I have a fix on the way after today. I'll add this one too which was discovered while working on that and will probably be fixed at the same time:

* [ ] [Fixup provenance for TrID results](https://github.com/richardlehane/siegfried/issues/160) (UX/reporting).

**Comment by ross-spencer** [comment]
Reflecting the past discussion https://github.com/richardlehane/siegfried/pull/152 is not looking likely for this release (marked in strike-thorough above) and its benefit still needs to be looked at in more detail by the team. The code still needs to be re-shaped too which I may do as an exercise either way.

**Comment by ross-spencer** [comment]
An update on my end is that the fix for https://github.com/richardlehane/siegfried/issues/153 is looking okay so I've submitted that as a proper PR. 

The provenance field issue ross-spencer/WikiDP-issues#4 looks okay, but I'll probably rebase that once the fix for #153 is merged. I am also conscious that while I am removing part of the code I haven't improved the test coverage and so have in the back of my mind to look at a unit test or two if they can be useful. 

Similarly, the new Wikibase provenance feature in #157 is looking good, but I have some review notes to address and I walked through the two libraries involved with connecting to Wikidata this weekend and wasn't happy with the coverage or some of the code discovered while testing. Spargo has been fixed up with [number of releases made](https://github.com/ross-spencer/spargo/releases). I suspect we'll use the latest one in the new update although it won't drop right in. The different releases may prove overkill and could have been avoided but provide backup options. 

I'd like to take some time this week to do the same with `wikiprov` the new sister package for adding the new permalink and revision history data. And then I have in mind a test for the Siegfried output that might compliment #153 as well.
